### PR TITLE
test: test_compat: run for more than one version

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: QA-514
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline please honey

**Review by lluiscampos — APPROVED** [review]
That was fast, thank you! :rocket: 

Please:
* Remove the changleog entry - the bug has not been released
* Review https://github.com/mendersoftware/integration/pull/2261 and optionally pull the commit here so that we test them together :+1:

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline pretty please honey

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline pretty please honey

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline pretty please honey

**Comment by merlin-northern** [comment]
> * Remove the changleog entry - the bug has not been released

fixed.

> * Review #2261 and optionally pull the commit here so that we test them together 👍

approved, maybe lets merge separately.

**Comment by merlin-northern** [comment]
@lluiscampos 
```
#cat /tmp/job.txt | grep -F docker-compose.compat | sed -e 's/.*\(docker-compose.compat-[^ ]*\).*/\1/' | sort | uniq # ent
docker-compose.compat-2.0.yml
docker-compose.compat-2.1.yml
docker-compose.compat-2.2.yml
docker-compose.compat-2.3.yml
docker-compose.compat-2.4.yml
docker-compose.compat-2.5.yml
docker-compose.compat-2.6.yml
docker-compose.compat-3.0.yml
docker-compose.compat-3.1.yml
docker-compose.compat-mender-3.2.yml
docker-compose.compat-mender-3.3.yml
docker-compose.compat-mender-3.4.yml
```

```
#cat /tmp/job.txt | grep -F docker-compose.compat | sed -e 's/.*\(docker-compose.compat-[^ ]*\).*/\1/' | sort | uniq # os
docker-compose.compat-2.0.yml
docker-compose.compat-2.1.yml
docker-compose.compat-2.2.yml
docker-compose.compat-2.3.yml
docker-compose.compat-2.4.yml
docker-compose.compat-2.5.yml
docker-compose.compat-2.6.yml
docker-compose.compat-3.0.yml
docker-compose.compat-3.1.yml
docker-compose.compat-mender-3.2.yml
docker-compose.compat-mender-3.3.yml
docker-compose.compat-mender-3.4.yml
```

File: testutils/infra/container_manager/docker_compose_manager.py [hunk 1]
```diff
@@ -403,8 +403,7 @@ def new_tenant_docker_client(self, name, tenant):
 class DockerComposeCompatibilitySetup(DockerComposeNamespace):
     def __init__(self, name, tag, enterprise=False):
         self._enterprise = enterprise
-        self.COMPAT_FILES_TEMPLATE[0] = self.COMPAT_FILES_TEMPLATE[0].format(tag=tag)
-        extra_files = self.COMPAT_FILES_TEMPLATE
+        extra_files = [self.COMPAT_FILES_TEMPLATE[0].format(tag=tag)]
```
Comment (lluiscampos) [inline testutils/infra/container_manager/docker_compose_manager.py:406]: `self.COMPAT_FILES_TEMPLATE` should not be an array, just a (format) string.

So I suggest we rename above to `COMPAT_FILES_TEMPLATE = COMPAT_FILES_ROOT + "/docker-compose.compat-{tag}.yml"` and here:
```suggestion
        extra_files = [self.COMPAT_FILES_TEMPLATE.format(tag=tag)]
```

**Review by lluiscampos — APPROVED** [review]
Looks good.

File: testutils/infra/container_manager/docker_compose_manager.py (hunk 1 above)
Comment (merlin-northern) [reply testutils/infra/container_manager/docker_compose_manager.py]: sure, fixed. (I wanted to stay aligned with the res of the class variables, but it is OK)

**Comment by merlin-northern** [comment]
@mender-test-bot start pipeline pretty please honey

**Comment by merlin-northern** [comment]
only known failures in tests/test_basic_integration.py::TestBasicIntegrationOpenSource::test_double_update_rofs  and a spurious failure in backend integration, all unrelated to the changes.

**Comment by merlin-northern** [comment]
I also checked 4 times that the different versions are tested.

**Comment by merlin-northern** [comment]
@mender-test-bot cherry-pick to:
 * 3.3.x
 * 3.4.x
please

**Pull request merged**